code/game/machinery/firealarm.dm

```diff
@@ -419,3 +425,37 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/firealarm, 26)
 	if (!party_overlay)
 		party_overlay = iconstate2appearance('icons/turf/areas.dmi', "party")
 	area.add_overlay(party_overlay)
+
+/obj/item/circuit_component/firealarm
+	display_name = "Fire Alarm"
+	desc = "Allows you to interface with the Fire Alarm."
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL
+
+	/////When the firealarm is used
+	var/datum/port/output/is_on
+
+	var/obj/machinery/firealarm/attached_alarm
+
+/obj/item/circuit_component/firealarm/populate_ports()
+	is_on = add_output_port("Active", PORT_TYPE_NUMBER)
+
+/obj/item/circuit_component/firealarm/register_usb_parent(atom/movable/parent)
+	. = ..()
+	if(istype(parent, /obj/machinery/firealarm))
+		attached_alarm = parent
+		RegisterSignal(parent, COMSIG_FIREALARM_USED, .proc/on_firealarm_used)
```

Comment:
if you have two separate signals here, you don't need an if statement.

Reply:
Agree